server/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -272,11 +266,40 @@ protected void configure() {
                 = MapBinder.newMapBinder(binder(), ActionType.class, TransportAction.class);
         for (ActionHandler<?, ?> action : actions.values()) {
             // bind the action as eager singleton, so the map binder one will reuse it
-            bind(action.getTransportAction()).asEagerSingleton();
-            transportActionsBinder.addBinding(action.getAction()).to(action.getTransportAction()).asEagerSingleton();
-            for (Class<?> supportAction : action.getSupportTransportActions()) {
+            bind(action.transportAction()).asEagerSingleton();
+            transportActionsBinder.addBinding(action.action()).to(action.transportAction()).asEagerSingleton();
+            for (Class<?> supportAction : action.supportTransportActions()) {
                 bind(supportAction).asEagerSingleton();
             }
         }
     }
+
+    /**
+     * Create a record of an action, the {@linkplain TransportAction} that handles it, and any supporting {@linkplain TransportActions}
+     * that are needed by that {@linkplain TransportAction}.
+     */
+    private record ActionHandler<Request extends TransportRequest, Response extends TransportResponse>(
+        ActionType<Response> action, Class<? extends TransportAction<Request, Response>> transportAction,
+        Class<?>... supportTransportActions) {
+
+        @Override
+        public String toString() {
+            StringBuilder b = new StringBuilder().append(action.name()).append(" is handled by ").append(transportAction.getName());
+            if (supportTransportActions.length > 0) {
+                b.append('[').append(Strings.arrayToCommaDelimitedString(supportTransportActions)).append(']');
+            }
+            return b.toString();
+        }
+
+        @Override
+        public boolean equals(Object obj) {
```

Comment:
Is this needed despite the record proving a default equals?

Reply:
probably not, checking and will remove.